tests/integration/tools/commands/test_venv.py

```diff
@@ -265,3 +265,75 @@ def colors_package_path(venv_dir):
     assert canonical_venv_listing == recursive_listing(venv_directory)
     venv = assert_app(venv_directory)
     assert canonical_venv_hash == CacheHelper.dir_hash(venv.site_packages_dir)
+
+
+def test_non_hermetic_issue_2004(tmpdir):
+    # type: (Any) -> None
+
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    pylint_pex = os.path.join(str(tmpdir), "pylint.pex")
+    run_pex_command(
+        args=[
+            "-m",
+            "pylint",
+            "pylint==2.15.8",
```

Comment:
Hrm, CI failing (at least) because this version isn't available for Python 2.7. Will see if I can make this work across versions.